src/main.js

```diff
@@ -166,46 +172,58 @@ async function shuttingDown(eventType, err) {
 		process.exit();
 	}
 }
-module.exports.shuttingDown = shuttingDown;
 
-const commandFiles = fs.readdirSync(path.join(__dirname, 'commands')).filter(file => file.endsWith('.js'));
-for (const file of commandFiles) {
+const locales = new Collection();
+const localeFolders = readdirSync(getAbsoluteFileURL(import.meta.url, ['..', 'locales']));
+for await (const folder of localeFolders) {
+	const localeFiles = readdirSync(getAbsoluteFileURL(import.meta.url, ['..', 'locales', folder]));
+	const localeData = {};
+	for await (const file of localeFiles) {
+		const locale = await import(getAbsoluteFileURL(import.meta.url, ['..', 'locales', folder, file]));
+		localeData[file.split('.')[0].toUpperCase()] = locale.default;
+	}
+	locales.set(folder, localeData);
+}
+setLocales(locales);
+
+const commandFiles = readdirSync(getAbsoluteFileURL(import.meta.url, ['commands'])).filter(file => file.endsWith('.js'));
+for await (const file of commandFiles) {
 	/** @type {{data: import('@discordjs/builders').SlashCommandBuilder, checks: string[], permissions: {user: string[], bot: string[]}, execute(interaction: import('discord.js').CommandInteraction): Promise<void>}} */
-	const command = require(path.join(__dirname, 'commands', file));
-	bot.commands.set(command.data.name, command);
+	const command = await import(getAbsoluteFileURL(import.meta.url, ['commands', file]));
+	bot.commands.set(command.default.data.name, command.default);
 }
 
-const componentsFolders = fs.readdirSync(path.join(__dirname, 'components'));
-for (const folder of componentsFolders) {
-	const componentFiles = fs.readdirSync(path.join(__dirname, 'components', folder)).filter(file => file.endsWith('.js'));
-	for (const file of componentFiles) {
-		const component = require(path.join(__dirname, 'components', folder, file));
+const componentsFolders = readdirSync(getAbsoluteFileURL(import.meta.url, ['components']));
+for await (const folder of componentsFolders) {
```

Comment:
I think the `await` here only matters for the second `for...of` loop inside this.
```suggestion
for (const folder of componentsFolders) {
```